admin/src/App.js

```diff
@@ -0,0 +1,54 @@
+import {
+  BrowserRouter as Router,
+  Route,
+  Routes,
+  Navigate,
+} from "react-router-dom";
+import PrivateRoute from "./components/HOC/PrivateRoutes";
+import { useDispatch, useSelector } from "react-redux";
+
+import Home from "./containers/Home";
+import Products from "./containers/Products";
+import Login from "./containers/Login";
+import { isUserLoggedIn } from "./actions";
+
+import "./App.scss";
+import { useEffect } from "react";
+
+function App() {
+  const dispatch = useDispatch();
+  const auth = useSelector((state) => state.auth);
```

Comment:
```suggestion
  const { isAuthenticate } = useSelector((state) => state.auth);
```
boolean thì để dạng is... has... anh nhé
https://www.samanthaming.com/tidbits/34-better-boolean-variable-names/